### ISSUE TITLE
Allow for a busy-out of the ECS-Agent so it accepts no new tasks

### ISSUE THREAD

**Issue by danbf** [aws/amazon-ecs-agent#649]
Right now trying to cleanly shutdown an ECS host is a problem.  If we could busy out an ecs-agent in combination with the ECS scheduler starting a new task instance prior to stopping an existing instance when a stop task is issued would allow for a clean shutdown of ECS hosts with no service outages.

Right now either a call to `stop task` or `deregister-container-instance` results in the elb connections to the existing task instances being shutdown before the new task instances are registered and working on an ELB, so there is an interruption of service when a ASG cluster is rolled via cloudformation. 

This is related to https://github.com/aws/amazon-ecs-agent/issues/130

**Comment by schickling** [comment]
This would be super helpful! 👍

**Comment by cbbarclay** [comment]
Today we launched [Container Instance Draining](http://docs.aws.amazon.com/AmazonECS/latest/developerguide/container-instance-draining.html) to address this use case.

**Comment by danbf** [comment]
@cbbarclay yup that almost entirely scratches my itch.  

we just need to be clever about the tasks that truly need to be run only once which container instance draining would run twice on a cluster for a short time.  fortunately, most of mine can tolerate that.

thanks much
![deb72188-c2b0-11e6-8b82-2c355940cec1](https://cloud.githubusercontent.com/assets/5041307/22306317/ce1e6076-e30c-11e6-97a9-ddf93bc22bfe.gif)
